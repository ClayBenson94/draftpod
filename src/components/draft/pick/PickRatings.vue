<script>

import { Drop } from 'vue-drag-drop'
import * as selectors from '@/store/modules/draft/selectors'

export default {
  name: 'PickRatings',

  filters: {
    rating: function(value) {
      return (Math.round( value * 10 ) / 10).toFixed(1);
    }
  },

  components: {
    Drop
  },

  props: {
    pick_ratings: {
      type: Array,
      required: true
    }
  },

  inject: [
    'touchDragManager'
  ],

  mounted() {
    this.touchDragManager.registerDropTarget({
      element: this.$el,
      handlers: {
        onMove: this.handleDragover,
      }
    });
  },

  beforeDestroy() {
    this.touchDragManager.unregisterDropTarget(this.$el);
  },

  methods: {
    handleDragover(data) {
      this.$emit('pick-dragged', data);
      return true;
    },

    card_colors: function(card) {
      return selectors.cardColors([card], true).filter((color) => color.count > 0);
    }
  },
}

</script>


<template>
  <Drop 
    v-if="pick_ratings.length > 0" 
    @dragover="handleDragover(...arguments)">
    <table class="ratings-table table table-sm">
      <thead class="">
        <th>Card</th>
        <th>Colors</th>
        <th>Rating</th>
        <th>Color Bonus</th>
        <th>Pick Rating</th>
      </thead>
      <tbody>
        <tr 
          v-for="pick_rating in pick_ratings" 
          :key="pick_rating.card.id">
          <td width="15%">{{ pick_rating.card.name }}</td>
          <td width="20%">
            <img 
              v-for="color in card_colors(pick_rating.card)" 
              :key="color.name"
              :src="color.img" 
              class="mana-color">
          </td>
          <td width="15%">{{ pick_rating.base_rating | rating }}</td>
          <td 
            v-if="pick_rating.color_bonus > 0" 
            width="15%">+{{ pick_rating.color_bonus | rating }}</td>
          <td 
            v-else 
            width="15%" 
            class="no-color-bonus">&mdash;</td>
          <td width="15%">{{ pick_rating.rating | rating }}</td>
        </tr>
      </tbody>
    </table>



    <p class="ratings-note">
      Card ratings provide a rough idea of the relative power level between cards, 
      but don't necessarily reflect optimal draft selections. The pick rating displayed reflects a baseline
      card rating plus a bonus for matching the color of the deck (the bonuses are lower in pack 1). Ratings don't reflect
      draft signals, deck synergies, mana curve, and creatue/spell ratio&mdash;these considerations will often result in a
      pick different than the highest rated card. Ratings are based on the community P1P1 evaluations provided at 
      <a 
        target="_blank" 
        href="https://draftaholicsanonymous.com/">https://draftaholicsanonymous.com</a>.
    </p>


  </Drop>

</template>

<style>


.ratings-table {
  font-size: 0.8rem;
}

.ratings-note {
  font-size: 0.8rem;
  margin-left: 2px;
  margin-right: 2px;
  margin-top: 0;
  margin-bottom: 10px;
  padding: 12px;
  padding-top: 8px;
  border: 1px solid rgba(0, 0, 0, 0.1);
  border-radius: 0.25rem;
  background-color: rgb(232,232,232);
}

.ratings-note a {
  color: inherit;
}

.ratings-table .mana-color {
  width: 16px;
  margin-right: 5px;
}

.ratings-table th, .ratings-table td {
  text-align: center;
  border-top: 1px solid rgba(0,0,0,0.1);
}

.ratings-table thead th {
  border-bottom: none;
  font-weight: 600;
  background-color: rgb(232,232,232);
}

.ratings-table th:first-of-type, .ratings-table td:first-of-type {
  text-align: left;
}

.ratings-table .no-color-bonus {
  color: rgba(0,0,0,0.2);
}

</style>

