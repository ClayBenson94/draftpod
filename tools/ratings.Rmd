---
title: "Rating Distributions"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

source("download_ratings.R")
```

```{r}
plot_channel_fireball_ratings <- function(set) {
  ratings <- as.numeric(channel_fireball_ratings(set)$rating)
  hist(ratings, main = set)
}
```

```{r}
all_ratings <- c(
  as.numeric(channel_fireball_ratings("dom")$rating),
  as.numeric(channel_fireball_ratings("m19")$rating),
  as.numeric(channel_fireball_ratings("grn")$rating),
  as.numeric(channel_fireball_ratings("rix")$rating),
  as.numeric(channel_fireball_ratings("xln")$rating)
)
hist(all_ratings, main = "All")

freq <- table(all_ratings)
pct <- prop.table(freq)
rating_dist <- data.frame(
  rating = row.names(freq),
  freq = as.numeric(freq),
  pct = round(as.numeric(pct) * 100, digits = 2)
)
rating_dist
```



```{r}
plot_channel_fireball_ratings("dom")
plot_channel_fireball_ratings("m19")
plot_channel_fireball_ratings("grn")
plot_channel_fireball_ratings("rix")
plot_channel_fireball_ratings("xln")
```

