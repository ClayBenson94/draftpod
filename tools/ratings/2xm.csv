"id","name","rating"
489674,"Karn Liberated",4.5
489675,"Alabaster Mage",1
489676,"Ancestral Blade",2
489677,"Angel of the Dawn",1.5
489678,"Archangel of Thune",4.5
489679,"Auriok Salvagers",2
489680,"Austere Command",4
489681,"Avacyn, Angel of Hope",4
489682,"Blade Splicer",4
489683,"Boon Reflection",0.5
489684,"Council's Judgment",3
489685,"Crib Swap",3
489686,"Crusader of Odric",1
489687,"Ethersworn Canonist",1.5
489688,"Fencing Ace",1
489689,"Flickerwisp",3
489690,"Fortify",0
489691,"Glint-Sleeve Artisan",1.5
489692,"Kemba, Kha Regent",3
489693,"Land Tax",3
489694,"Leonin Abunas",2
489695,"Master Splicer",3.5
489696,"Myrsmith",3
489697,"Open the Vaults",2
489698,"Path to Exile",4
489699,"Puresteel Paladin",3.5
489700,"Remember the Fallen",1
489701,"Revoke Existence",2.5
489702,"Sanctum Gargoyle",1.5
489703,"Sanctum Spirit",1
489704,"Stoneforge Mystic",3.5
489705,"Stonehewer Giant",3.5
489706,"Strength of Arms",1.5
489707,"Tempered Steel",3
489708,"Thraben Inspector",2
489709,"Topple the Statue",3
489710,"Valor in Akros",0.5
489711,"Valorous Stance",1.5
489712,"Wrath of God",4.5
489713,"Apprentice Wizard",1.5
489714,"Arcum Dagsson",3
489715,"Argivian Restoration",1
489716,"Braids, Conjurer Adept",1.5
489717,"Brainstorm",1.5
489718,"Cloudreader Sphinx",1
489719,"Corridor Monitor",1
489720,"Cyclonic Rift",3.5
489721,"Deepglow Skate",2
489722,"Esperzoa",3.5
489723,"Faerie Mechanist",1.5
489724,"Force of Will",3.5
489725,"Frogify",1.5
489726,"Grand Architect",3.5
489727,"Hinder",2
489728,"Inkwell Leviathan",2.5
489729,"Jace, the Mind Sculptor",4
489730,"Master of Etherium",3.5
489731,"Master Transmuter",3.5
489732,"Metallic Rebuke",2.5
489733,"Parasitic Strix",2
489734,"Phyrexian Metamorph",3.5
489735,"Pongify",2.5
489736,"Relic Runner",1
489737,"Reshape",2.5
489738,"Riddlesmith",1.5
489739,"Rush of Knowledge",1
489740,"Sentinel of the Pearl Trident",1.5
489741,"Serra Sphinx",2.5
489742,"Sift",2
489743,"Steel Sabotage",2.5
489744,"Thirst for Knowledge",3
489745,"Thought Reflection",1
489746,"Treasure Mage",3
489747,"Vedalken Infuser",1
489748,"Well of Ideas",1.5
489749,"Ad Nauseam",1
489750,"Beacon of Unrest",2.5
489751,"Bone Picker",2
489752,"Cast Down",3
489753,"Costly Plunder",0.5
489754,"Dark Confidant",4
489755,"Death's Shadow",3
489756,"Defiant Salvager",2
489757,"Dire Fleet Hoarder",1.5
489758,"Disciple of Bolas",3
489759,"Disciple of the Vault",2
489760,"Divest",1
489761,"Doomed Necromancer",3
489762,"Dread Return",2
489763,"Driver of the Dead",0.5
489764,"Drown in Sorrow",2.5
489765,"Executioner's Capsule",3.5
489766,"Fatal Push",4
489767,"Geth, Lord of the Vault",4
489768,"Glaze Fiend",1.5
489769,"Heartless Pillage",0.5
489770,"Magus of the Abyss",3
489771,"Magus of the Will",2
489772,"Morkrut Banshee",3
489773,"Oubliette",3
489775,"Painsmith",1.5
489776,"Ravenous Trap",0
489777,"Salvage Titan",3
489778,"Silumgar Scavenger",1.5
489779,"Skirsdag High Priest",3
489780,"Skithiryx, the Blight Dragon",4.5
489781,"Supernatural Stamina",0
489782,"Thoughtseize",3.5
489783,"Toxic Deluge",4.5
489784,"Twisted Abomination",2
489785,"Vampire Hexmage",1.5
489786,"Wound Reflection",2.5
489787,"Abrade",3
489788,"Balduvian Rage",1
489789,"Battle-Rattle Shaman",1.5
489790,"Blasphemous Act",3.5
489791,"Blood Moon",1.5
489792,"Bloodshot Trainee",1.5
489793,"Brimstone Volley",3
489794,"Cathartic Reunion",1
489795,"Cragganwick Cremator",1
489796,"Dismantle",1.5
489797,"Dualcaster Mage",3
489798,"Galvanic Blast",3.5
489799,"Goblin Gaveleer",0.5
489800,"Goblin Guide",3
489801,"Godo, Bandit Warlord",3.5
489802,"Grim Lavamancer",3
489803,"Heat Shimmer",1.5
489804,"Imperial Recruiter",3.5
489805,"Ion Storm",1.5
489806,"Kazuul's Toll Collector",2
489807,"Kuldotha Flamefiend",3
489808,"Lightning Axe",3
489809,"Mana Echoes",1.5
489810,"Orcish Vandal",1.5
489811,"Pyrewild Shaman",2
489812,"Rage Reflection",1.5
489813,"Rapacious Dragon",2
489814,"Ravenous Intruder",1
489815,"Rolling Earthquake",3.5
489816,"Salivating Gremlins",1
489817,"Skinbrand Goblin",0.5
489818,"Sneak Attack",3.5
489819,"Temur Battle Rage",1.5
489820,"Thopter Engineer",3
489821,"Trash for Treasure",1
489822,"Tuktuk the Explorer",3.5
489823,"Weapon Surge",1
489824,"Ancient Stirrings",1.5
489825,"Avenger of Zendikar",4.5
489826,"Awakening Zone",3
489827,"Bloodbriar",1.5
489828,"Bloodspore Thrinax",3
489829,"Champion of Lambholt",3.5
489830,"Chatter of the Squirrel",0.5
489831,"Chord of Calling",3
489832,"Clear Shot",2.5
489833,"Conclave Naturalists",3
489834,"Crop Rotation",2
489835,"Crushing Vines",1.5
489836,"Death-Hood Cobra",1
489837,"Doubling Season",3.5
489838,"Elvish Aberration",1.5
489839,"Enlarge",1
489840,"Exploration",2.5
489841,"Fierce Empath",3.5
489842,"Gelatinous Genesis",3
489843,"Greater Good",3
489844,"Heartbeat of Spring",1
489845,"Invigorate",1
489846,"Kozilek's Predator",2.5
489847,"Liege of the Tangle",3
489848,"Mana Reflection",1.5
489849,"Might of the Masses",0
489850,"Noble Hierarch",3.5
489851,"Reclamation Sage",3.5
489852,"Shamanic Revelation",2.5
489853,"Skullmulcher",2.5
489854,"Sylvan Might",1
489855,"Terastodon",3
489856,"Thragtusk",3.5
489857,"Ulvenwald Mysteries",3
489858,"Vengevine",3
489859,"Veteran Explorer",1
489860,"Whisperer of the Wilds",2.5
489861,"Woodland Champion",1
489862,"Arixmethes, Slumbering Isle",3.5
489863,"Atraxa, Praetors' Voice",3
489864,"Baleful Strix",3.5
489865,"Breya, Etherium Shaper",3.5
489866,"Brudiclad, Telchor Engineer",2.5
489867,"Deathreap Ritual",2.5
489868,"Falkenrath Aristocrat",3
489869,"Fulminator Mage",3
489870,"Geist of Saint Traft",4
489871,"Ghor-Clan Rampager",2
489872,"Glassdust Hulk",3
489873,"Hanna, Ship's Navigator",3
489874,"Hidden Stockpile",3
489875,"Izzet Charm",3
489876,"Jhoira, Weatherlight Captain",3.5
489877,"Kaalia of the Vast",2
489878,"Karrthus, Tyrant of Jund",3
489879,"Maelstrom Nexus",0
489880,"Maelstrom Pulse",3.5
489881,"Manamorphose",3
489882,"Mazirek, Kraul Death Priest",3.5
489883,"Meddling Mage",2.5
489884,"Merciless Eviction",3
489885,"Progenitor Mimic",3.5
489886,"Rhys the Redeemed",3.5
489887,"Riku of Two Reflections",3.5
489888,"Savageborn Hydra",3.5
489889,"The Scarab God",4
489890,"Selesnya Guildmage",2.5
489891,"Sen Triplets",4
489892,"Sharuum the Hegemon",3
489893,"Sphinx Summoner",3.5
489894,"Swiftblade Vindicator",3.5
489895,"Thopter Foundry",4
489896,"Time Sieve",0
489898,"Vexing Shusher",3
489899,"Vish Kal, Blood Arbiter",3
489900,"Voice of Resurgence",3.5
489901,"Weapons Trainer",3
489902,"Yavimaya's Embrace",1.5
489903,"Accomplished Automaton",1.5
489904,"Adaptive Automaton",3
489905,"Basalt Monolith",3
489906,"Basilisk Collar",4
489907,"Batterskull",4
489908,"Blightsteel Colossus",3.5
489909,"Bosh, Iron Golem",3
489910,"Cathodion",1.5
489911,"Chief of the Foundry",3
489912,"Chromatic Star",2
489913,"Chrome Mox",3.5
489914,"Clone Shell",2.5
489915,"Cogwork Assembler",2
489916,"Conjurer's Closet",3
489917,"Coretapper",2
489918,"Cranial Plating",3.5
489919,"Culling Dais",1
489920,"Darksteel Axe",2.5
489921,"Darksteel Forge",1.5
489922,"Duplicant",3.5
489923,"Eager Construct",1
489924,"Endless Atlas",3
489925,"Engineered Explosives",3
489926,"Ensnaring Bridge",2
489927,"Everflowing Chalice",3
489928,"Expedition Map",3
489929,"Flayer Husk",2
489930,"Gleaming Barrier",0
489931,"Golem Artisan",3
489932,"Golem-Skin Gauntlets",0.5
489933,"Hammer of Nazahn",4
489934,"Ichor Wellspring",2.5
489935,"Iron Bully",0.5
489936,"Iron League Steed",1
489937,"Isochron Scepter",3.5
489938,"Jhoira's Familiar",1.5
489939,"Kuldotha Forgemaster",1.5
489940,"Lightning Greaves",3.5
489941,"Lux Cannon",3.5
489942,"Magnifying Glass",2.5
489943,"Mana Crypt",4.5
489944,"Masterwork of Ingenuity",3
489945,"Mesmeric Orb",1.5
489946,"Metalspinner's Puzzleknot",1.5
489947,"Mishra's Bauble",3
489948,"Mox Opal",3
489949,"Myr Battlesphere",3.5
489950,"Myr Retriever",2.5
489951,"O-Naginata",2
489952,"Oblivion Stone",3.5
489953,"Peace Strider",1
489954,"Pentad Prism",2.5
489955,"Phyrexian Revoker",2.5
489956,"Pyrite Spellbomb",3
489957,"Ratchet Bomb",3
489958,"Sandstone Oracle",2
489959,"Sculpting Steel",3.5
489960,"Sickleslicer",1
489961,"Skinwing",0.5
489962,"Spellskite",3.5
489963,"Sphinx of the Guildpact",1.5
489964,"Springleaf Drum",1
489965,"Sundering Titan",3
489966,"Sunforger",3.5
489967,"Surge Node",0
489968,"Sword of Body and Mind",4.5
489969,"Sword of Feast and Famine",4
489970,"Sword of Fire and Ice",5
489971,"Sword of Light and Shadow",4
489972,"Sword of the Meek",3.5
489973,"Sword of War and Peace",4.5
489974,"Throne of Geth",1
489975,"Treasure Keeper",2
489976,"Trinisphere",2
489977,"Tumble Magnet",1
489978,"Vulshok Gauntlets",0
489979,"Walking Ballista",4
489980,"Welding Jar",1
489981,"Wurmcoil Engine",4.5
489982,"Academy Ruins",2.5
489983,"Ash Barrens",2
489984,"Blinkmoth Nexus",3
489985,"Buried Ruin",1.5
489986,"Cascade Bluffs",2
489987,"Dark Depths",3
489988,"Darksteel Citadel",0.5
489989,"Fetid Heath",2
489990,"Fire-Lit Thicket",1
489991,"Flooded Grove",2.5
489992,"Glimmervoid",1.5
489993,"Graven Cairns",1.5
489994,"High Market",1
489995,"Maze of Ith",3
489996,"Mishra's Factory",2
489997,"Mystic Gate",2
489998,"Rugged Prairie",2
489999,"Sunken Ruins",2.5
490000,"Thespian's Stage",1
490001,"Twilight Mire",1
490002,"Urza's Mine",1
490003,"Urza's Power Plant",0
490004,"Urza's Tower",0.5
490005,"Wooded Bastion",2
