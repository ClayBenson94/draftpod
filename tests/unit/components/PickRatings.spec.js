

import { mount } from '@vue/test-utils'
import PickRatings from '../../../src/components/draft/pick/PickRatings.vue'

import ratings from '../../data/ratings.json'

import providers from '../../util/providers'

describe('PickRatings.vue', () => {

  test('displays ratings', () => {
    const wrapper = mount(PickRatings, {
      propsData: { pick_ratings: ratings },
      provide: providers
    });
    const ratingRows = wrapper.findAll('.ratings-table tbody>tr');
    expect(ratingRows).toHaveLength(ratings.length);

    // do a snapshot test
    expect(wrapper.element).toMatchSnapshot();
  });

});

