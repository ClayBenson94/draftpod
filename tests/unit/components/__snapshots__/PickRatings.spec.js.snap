// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickRatings.vue displays ratings 1`] = `
<div>
  <table
    class="ratings-table table table-sm"
  >
    <thead>
      <th>
        Card
      </th>
       
      <th>
        Colors
      </th>
       
      <th>
        Rating
      </th>
       
      <th>
        Color Bonus
      </th>
       
      <th>
        Pick Rating
      </th>
    </thead>
     
    <tbody>
      <tr>
        <td
          width="15%"
        >
          
          Sphinx of Foresight
        
        </td>
         
        <td
          width="20%"
        >
          <img
            class="mana-color"
            src="/images/mana-blue.svg"
          />
        </td>
         
        <td
          width="15%"
        >
          
          4.0
        
        </td>
         
        <td
          class="no-color-bonus"
          width="15%"
        >
          
          —
        
        </td>
         
        <td
          width="15%"
        >
          
          4.0
        
        </td>
      </tr>
      <tr>
        <td
          width="15%"
        >
          
          Collision // Colossus
        
        </td>
         
        <td
          width="20%"
        >
          <img
            class="mana-color"
            src="/images/mana-red.svg"
          />
          <img
            class="mana-color"
            src="/images/mana-green.svg"
          />
        </td>
         
        <td
          width="15%"
        >
          
          3.5
        
        </td>
         
        <td
          class="no-color-bonus"
          width="15%"
        >
          
          —
        
        </td>
         
        <td
          width="15%"
        >
          
          3.5
        
        </td>
      </tr>
      <tr>
        <td
          width="15%"
        >
          
          Gates Ablaze
        
        </td>
         
        <td
          width="20%"
        >
          <img
            class="mana-color"
            src="/images/mana-red.svg"
          />
        </td>
         
        <td
          width="15%"
        >
          
          3.0
        
        </td>
         
        <td
          class="no-color-bonus"
          width="15%"
        >
          
          —
        
        </td>
         
        <td
          width="15%"
        >
          
          3.0
        
        </td>
      </tr>
      <tr>
        <td
          width="15%"
        >
          
          Incubation // Incongruity
        
        </td>
         
        <td
          width="20%"
        >
          <img
            class="mana-color"
            src="/images/mana-green.svg"
          />
          <img
            class="mana-color"
            src="/images/mana-blue.svg"
          />
        </td>
         
        <td
          width="15%"
        >
          
          2.5
        
        </td>
         
        <td
          class="no-color-bonus"
          width="15%"
        >
          
          —
        
        </td>
         
        <td
          width="15%"
        >
          
          2.5
        
        </td>
      </tr>
      <tr>
        <td
          width="15%"
        >
          
          Rakdos Guildgate
        
        </td>
         
        <td
          width="20%"
        >
          <img
            class="mana-color"
            src="/images/mana-black.svg"
          />
          <img
            class="mana-color"
            src="/images/mana-red.svg"
          />
        </td>
         
        <td
          width="15%"
        >
          
          2.5
        
        </td>
         
        <td
          class="no-color-bonus"
          width="15%"
        >
          
          —
        
        </td>
         
        <td
          width="15%"
        >
          
          2.5
        
        </td>
      </tr>
      <tr>
        <td
          width="15%"
        >
          
          Ghor-Clan Wrecker
        
        </td>
         
        <td
          width="20%"
        >
          <img
            class="mana-color"
            src="/images/mana-red.svg"
          />
        </td>
         
        <td
          width="15%"
        >
          
          2.0
        
        </td>
         
        <td
          class="no-color-bonus"
          width="15%"
        >
          
          —
        
        </td>
         
        <td
          width="15%"
        >
          
          2.0
        
        </td>
      </tr>
      <tr>
        <td
          width="15%"
        >
          
          Impassioned Orator
        
        </td>
         
        <td
          width="20%"
        >
          <img
            class="mana-color"
            src="/images/mana-white.svg"
          />
        </td>
         
        <td
          width="15%"
        >
          
          2.0
        
        </td>
         
        <td
          class="no-color-bonus"
          width="15%"
        >
          
          —
        
        </td>
         
        <td
          width="15%"
        >
          
          2.0
        
        </td>
      </tr>
      <tr>
        <td
          width="15%"
        >
          
          Senate Courier
        
        </td>
         
        <td
          width="20%"
        >
          <img
            class="mana-color"
            src="/images/mana-white.svg"
          />
          <img
            class="mana-color"
            src="/images/mana-blue.svg"
          />
        </td>
         
        <td
          width="15%"
        >
          
          2.0
        
        </td>
         
        <td
          class="no-color-bonus"
          width="15%"
        >
          
          —
        
        </td>
         
        <td
          width="15%"
        >
          
          2.0
        
        </td>
      </tr>
      <tr>
        <td
          width="15%"
        >
          
          Storm Strike
        
        </td>
         
        <td
          width="20%"
        >
          <img
            class="mana-color"
            src="/images/mana-red.svg"
          />
        </td>
         
        <td
          width="15%"
        >
          
          1.5
        
        </td>
         
        <td
          class="no-color-bonus"
          width="15%"
        >
          
          —
        
        </td>
         
        <td
          width="15%"
        >
          
          1.5
        
        </td>
      </tr>
      <tr>
        <td
          width="15%"
        >
          
          Tenth District Veteran
        
        </td>
         
        <td
          width="20%"
        >
          <img
            class="mana-color"
            src="/images/mana-white.svg"
          />
        </td>
         
        <td
          width="15%"
        >
          
          1.5
        
        </td>
         
        <td
          class="no-color-bonus"
          width="15%"
        >
          
          —
        
        </td>
         
        <td
          width="15%"
        >
          
          1.5
        
        </td>
      </tr>
      <tr>
        <td
          width="15%"
        >
          
          Dead Revels
        
        </td>
         
        <td
          width="20%"
        >
          <img
            class="mana-color"
            src="/images/mana-black.svg"
          />
        </td>
         
        <td
          width="15%"
        >
          
          1.5
        
        </td>
         
        <td
          class="no-color-bonus"
          width="15%"
        >
          
          —
        
        </td>
         
        <td
          width="15%"
        >
          
          1.5
        
        </td>
      </tr>
      <tr>
        <td
          width="15%"
        >
          
          Expose to Daylight
        
        </td>
         
        <td
          width="20%"
        >
          <img
            class="mana-color"
            src="/images/mana-white.svg"
          />
        </td>
         
        <td
          width="15%"
        >
          
          1.0
        
        </td>
         
        <td
          class="no-color-bonus"
          width="15%"
        >
          
          —
        
        </td>
         
        <td
          width="15%"
        >
          
          1.0
        
        </td>
      </tr>
      <tr>
        <td
          width="15%"
        >
          
          Watchful Giant
        
        </td>
         
        <td
          width="20%"
        >
          <img
            class="mana-color"
            src="/images/mana-white.svg"
          />
        </td>
         
        <td
          width="15%"
        >
          
          0.5
        
        </td>
         
        <td
          class="no-color-bonus"
          width="15%"
        >
          
          —
        
        </td>
         
        <td
          width="15%"
        >
          
          0.5
        
        </td>
      </tr>
      <tr>
        <td
          width="15%"
        >
          
          Clear the Mind
        
        </td>
         
        <td
          width="20%"
        >
          <img
            class="mana-color"
            src="/images/mana-blue.svg"
          />
        </td>
         
        <td
          width="15%"
        >
          
          0.5
        
        </td>
         
        <td
          class="no-color-bonus"
          width="15%"
        >
          
          —
        
        </td>
         
        <td
          width="15%"
        >
          
          0.5
        
        </td>
      </tr>
      <tr>
        <td
          width="15%"
        >
          
          Burn Bright
        
        </td>
         
        <td
          width="20%"
        >
          <img
            class="mana-color"
            src="/images/mana-red.svg"
          />
        </td>
         
        <td
          width="15%"
        >
          
          0.0
        
        </td>
         
        <td
          class="no-color-bonus"
          width="15%"
        >
          
          —
        
        </td>
         
        <td
          width="15%"
        >
          
          0.0
        
        </td>
      </tr>
    </tbody>
  </table>
   
  <p
    class="ratings-note"
  >
    
    Card ratings provide a rough idea of the relative power level between cards, 
    but don't necessarily reflect optimal draft selections. The pick rating displayed reflects a baseline
    card rating plus a bonus for matching the color of the deck (the bonuses are lower in pack 1). Ratings don't reflect
    draft signals, deck synergies, mana curve, and creatue/spell ratio—these considerations will often result in a
    pick different than the highest rated card. Ratings are based on the community P1P1 evaluations provided at 
    
    <a
      href="https://draftaholicsanonymous.com/"
      target="_blank"
    >
      
      https://draftaholicsanonymous.com
    
    </a>
    .
  
  </p>
</div>
`;
